src/main/java/org/sefglobal/scholarx/service/ProgramService.java

```diff
@@ -204,4 +204,62 @@ public Mentor applyAsMentor(long programId, long profileId, Mentor mentor)
         mentor.setState(EnrolmentState.PENDING);
         return mentorRepository.save(mentor);
     }
+
+    /**
+     * Retrieves the {@link Mentor} of a user if the user is a mentor
+     *
+     * @param programId which is the id of the {@link Program}
+     * @param profileId which is the id of the {@link Profile}
+     * @return {@link Mentor}
+     *
+     * @throws ResourceNotFoundException if the requesting {@link Mentor} doesn't exist
+     */
+    public Mentor getLoggedInMentor(long programId, long profileId)
+            throws ResourceNotFoundException {
+        Optional<Mentor> optionalMentor = mentorRepository.findByProfileIdAndProgramId(profileId, programId);
+        if (!optionalMentor.isPresent()) {
+            String msg = "Error, Mentor by profile id: " + profileId + " and " +
+                         "program id: " + programId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+        return optionalMentor.get();
+    }
+
+    /**
+     * Update the application and prerequisites of a {@link Mentor}
+     *
+     * @param profileId which is the Profile id of the {@link Mentor} to be updated
+     * @param programId which is the Program id of the {@link Mentor} to be updated
+     * @param mentor    with the application and prerequisites of the mentor to be updated
+     * @return the updated {@link Mentor}
+     *
+     * @throws ResourceNotFoundException is thrown if the {@link Mentor} doesn't exist
+     */
+    public Mentor updateMentorData(long profileId, long programId, Mentor mentor)
+            throws ResourceNotFoundException {
+        Optional<Mentor> optionalMentor = mentorRepository.findByProfileIdAndProgramId(profileId, programId);
+        if (!optionalMentor.isPresent()) {
+            String msg = "Error, Mentor by profile id: " + profileId + " and " +
+                         "program id: " + programId + " cannot be updated. " +
+                         "Mentor doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+
+        Mentor savedMentor = optionalMentor.get();
+        if (!(mentor.getApplication().isEmpty() || savedMentor.getApplication().equals(mentor.getApplication()))) {
+            if (EnrolmentState.PENDING.equals(savedMentor.getState())) {
+                savedMentor.setApplication(mentor.getApplication());
+            }
+        }
+        if (!(mentor.getPrerequisites().isEmpty() || savedMentor.getPrerequisites().equals(mentor.getPrerequisites()))) {
+            if (EnrolmentState.APPROVED.equals(savedMentor.getState()) || EnrolmentState.PENDING.equals(savedMentor.getState())) {
```

Comment:
Create `isHigherThanOrEqual` in the Enum
```suggestion
            if (EnrolmentState.APPROVED.isHigherThanOrEqual(savedMentor.getState()) {
```

Reply:
Added a method to the Enums